Add NodeLease e2e tests to gci-gce-alpha-features ci tests

wangzhen127 opened the pull request:
This PR adds NodeLease e2e tests to gci-gce-alpha-features ci tests

This is part of [KEP-0009](https://github.com/kubernetes/community/blob/master/keps/sig-node/0009-node-heartbeat.md), feature [#589](https://github.com/kubernetes/features/issues/589) and issue [#14733](https://github.com/kubernetes/kubernetes/issues/14733).

wangzhen127 commented:
/assign @krzyzacy 
/assign @yujuhong 
/cc @wojtek-t

wangzhen127 commented:
@krzyzacy and @yujuhong, when I run the following on my local workstation in k8s repo, the NodeLease tests fail.
```
KUBE_FEATURE_GATES=AllAlpha=true kubetest --build --up --test \
--provider=gce \
--gcp-project=zhenw-gke-dev \
--runtime-config=api/all=true \
--test_args="--ginkgo.focus=\[Feature:NodeLease\]"
```

I ssh onto the machines and find that:
- Master's kubelet is started with `--feature-gates=AllAlpha=true`
- Node's kubelet is started **without** `--feature-gates=AllAlpha=true`

I suspect that's the reason why the tests fail. Is there any way to make nodes also use the feature gate?

krzyzacy commented:
I saw the node-alpha-feature test has `--node-test-args=--feature-gates=AllAlpha=true` which passed into the node test runner, maybe we need to somehow mimic that in cluster e2e?

(I think you are the first one try to run a node feature on cluster e2e? so... punt to @yujuhong :-) )

wangzhen127 commented:
I just figured out that kubelet is using dynamic config and both mater and nodes are using AllAlpha. The failure was on the test side. I have fixed that.

Now I split this PR into 2:
1. This PR: Add NodeLease e2e tests to gci-gce-alpha-features ci tests (This can only be submitted after the e2e test fix is submitted.)
2. #9943: Add optional presubmit pull-kubernetes-e2e-gce-alpha-features tests (This can be submitted right away, and the e2e test can use this optional presubmit to check the tests are passing.)

/hold
Holding this PR until e2e tests are fixed.

wangzhen127 commented:
@krzyzacy, this is now unblocked. Can you take another look?

wangzhen127 commented:
/hold cancel
